hat/app/org/hatdex/hat/api/controllers/Authentication.scala

```diff
@@ -277,125 +326,111 @@ class Authentication @Inject() (
+                    // ???: return an AuthenticatorResult, why
                     result
                   }
                 case None =>
-                  Future.successful(
-                    Unauthorized(
-                      Json.toJson(
-                        ErrorMessage(
-                          "Password reset unauthorized",
-                          "No user matching token"
-                        )
-                      )
-                    )
-                  )
+                  Future.successful(noUserMatchingToken)
               }
           } else {
-            Future.successful(
-              Unauthorized(
-                Json.toJson(
-                  ErrorMessage(
-                    "Password reset unauthorized",
-                    "Only HAT owner can reset their password"
-                  )
-                )
-              )
-            )
+            Future.successful(onlyHatOwnerCanReset)
           }
+        }
         case Some(_) =>
           tokenService.consume(tokenId)
-          Future.successful(
-            Unauthorized(
-              Json.toJson(
-                ErrorMessage("Invalid Token", "Token expired or invalid")
-              )
-            )
-          )
+          Future.successful(expiredToken)
         case None =>
-          Future.successful(
-            Unauthorized(
-              Json.toJson(ErrorMessage("Invalid Token", "Token does not exist"))
-            )
-          )
+          Future.successful(invalidToken)
       }
     }
 
   /**
     * Sends an email to the owner with a link to claim the hat
+    * /control/v2/auth/claim
     */
+
+    // Role EmailVerified
   def handleClaimStart(lang: Option[String]): Action[ApiClaimHatRequest] =
     UserAwareAction.async(parsers.json[ApiClaimHatRequest]) {
       implicit request =>
-        implicit val language: Lang =
-          Lang.get(lang.getOrElse("en")).getOrElse(Lang.defaultLang)
-
+        
+        implicit val language: Lang = Lang.get(lang.getOrElse("en")).getOrElse(Lang.defaultLang)
         val claimHatRequest = request.body
+        // ???: Not sure how this is hydrated.
         val email = request.dynamicEnvironment.ownerEmail
-        val response = Ok(
-          Json.toJson(
-            SuccessResponse(
-              "You will shortly receive an email with claim instructions"
-            )
-          )
-        )
+        val response = Ok(Json.toJson(SuccessResponse("You will shortly receive an email with claim instructions")))
+        val scheme = if (request.secure) "https://" else "http://"
 
+        // (email, applicationId) in the body
+        // Match the mail to that of the ENV
+        // Look up the application (Is this in the HAT itself?  Not DEX)
         if (claimHatRequest.email == email) {
           usersService.listUsers
-            .map(_.find(_.roles.contains(Owner())))
+            .map(_.find(u -> (u.roles.contains(Owner()) && !(u.roles.contains(EmailVerified)))
```

Comment:
This line needed quite a few changes to compile:
```.map(_.find(u => (u.roles.contains(Owner()) && !(u.roles.contains(EmailVerified(""))))))```

I'm still not certain that `EmailVerified` will actually match unless the extra string value matches too. This might be a problem.

Reply:
We are tucking the date of verification in the extra field.  If that is unnecessary, we could switch to `EmailVerified` instead.